funccount.py

```diff
@@ -0,0 +1,84 @@
+import os
+import re
+import shutil
+import signal
+import subprocess as sp
+import threading
+import time
+
+class FunccountInvalidFunctionError(Exception):
+    pass
+
+class FunccountNotStartedError(Exception):
+    pass
+
+class Funccount:
+    def _sleep_and_run(self):
+        self.event.wait(self.sleep)
+
+        self.lock.acquire()
+        if not self.joined:
+            # Do not start funccount tool if stop() has already been called,
+            # otherwise we end up with a zombie thread
+            self.proc = sp.Popen(self.cmd, stdout=sp.PIPE, stderr=sp.PIPE,
+                                 text=True, start_new_session=True)
+        self.lock.release()
+
+    def __init__(self, funcs, executable='funccount', sleep=0):
```

Comment:
So, in debian, there are two funccounts with different names (that may behave slightly differently).

There's funccount-perf from the perf-tools-unstable package and funccount-bpfcc from the bpfcc-tools package.

There's no way to get a 'funccount' pure executable in Debian.

So I'd rather suggest, by default we search for all three possibilities and use whatever is installed.